nav2_regulated_pure_pursuit_controller/include/nav2_regulated_pure_pursuit_controller/regulated_pure_pursuit_controller.hpp

```diff
@@ -272,6 +286,12 @@ class RegulatedPurePursuitController : public nav2_core::Controller
   double rotate_to_heading_min_angle_;
   double goal_dist_tol_;
   bool allow_reversing_;
+  double robot_angle_;
+  bool rotating_;
+  double kp_angle_;
+  double max_linear_jerk_;
+  double max_angular_jerk_;
+  rclcpp::Time system_time_;
```

Comment:
This is reliably run at the `controller_frequency` rate, may that be substituted instead of keeping timers?

Reply:
True, it is run at mentioned frequency. I am not using timers, just tracking system time. The problem is, I need to reset some internal states in case goal has been cancelled or goal checker stopped calling `computeVelocityCommands`. There is no such callback that allows me to do that, so my solution is to check if `computeVelocityCommands` has not been called for 4 control loop durations. If that's true, we have to reset internal states since controller has been idle for a while. You can look at the logic here:
https://github.com/memristor/navigation2/blob/03cf93ce88499222ed5e599f1a5dad8e8572c50d/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L324-L344

Reply:
You should break that logic out into a separate function

Reply:
Would `setPlan` do that? E.g. its a new plan given. The 2 cases is that there's that delay you mention or its a replan during robot execution. 

The state reset if the time is too long seems like a better fit to be executed in `setPlan` than in this function